src/audio/src/src.c

```diff
@@ -406,15 +405,14 @@ static void src_2s(struct comp_dev *dev, const struct audio_stream __sparse_cach
 
 	s2.times = cd->param.stage2_times;
 	s2_blk_in = s2.times * cd->src.stage2->blk_in * nch;
-	s2_blk_out = s2.times * cd->src.stage2->blk_out * nch;
 	if (s2_blk_in > cd->sbuf_avail) {
 		s2.times = cd->sbuf_avail / (cd->src.stage2->blk_in * nch);
 		s2_blk_in = s2.times * cd->src.stage2->blk_in * nch;
-		s2_blk_out = s2.times * cd->src.stage2->blk_out * nch;
 		comp_dbg(dev, "s2.times = %d", s2.times);
 	}
 
 	/* Test if second stage can be run with default block length. */
+	s2_blk_out = s2.times * cd->src.stage2->blk_out * nch;
```

Comment:
same with above.

Reply:
Same, if you think I'm missing something please show a patch that I can test if it works.

Reply:
yes, missed s2.times is changing.